### PR TITLE
Add std:: qualifier to move() calls

### PR DESCRIPTION
**Pull request by Marenz**
Starting with clang-16, ``unqualified-std-cast-call`` will be enabled by default, which will warn for every unqualified ``move()`` call.

I wrote this code while trying to get the ems dockerimage update to work but ultimately decided for a different approach, namely adding the ``-Wno-unqualified-std-cast-call`` flag.

If we don't merge this PR we can only use clang-16 (and any later) with that flag.

So, if we want that, feel free to merge, if not, close.

### CODE REVIEW

File: libsolidity/analysis/NameAndTypeResolver.cpp [hunk 1]
```diff
@@ -450,7 +450,7 @@ vector<T const*> NameAndTypeResolver::cThreeMerge(list<list<T const*>>& _toMerge
 	{
 		for (auto it = _toMerge.begin(); it != _toMerge.end();)
 		{
-			it->remove(_candidate);
+			it->restd::move(_candidate);
```
Comment (cameel) [inline libsolidity/analysis/NameAndTypeResolver.cpp]: Looks like your search&replace regex was a bit too weak :)
```suggestion
			it->remove(_candidate);
```

File: libsolidity/formal/SymbolicState.cpp [hunk 2]
```diff
@@ -114,13 +114,13 @@ void SymbolicState::transfer(smtutil::Expression _from, smtutil::Expression _to,
...
 	addBalance(_from, 0 - _value);
-	addBalance(_to, move(_value));
+	addBalance(_to, std::move(_value));
 	unsigned indexAfter = m_state.index();
 	solAssert(indexAfter > indexBefore, "");
 	m_state.newVar();
 	/// Do not apply the transfer operation if _from == _to.
 	auto newState = smtutil::Expression::ite(
-		move(_from) == move(_to),
+		std::move(_from) == std::move(_to),
```
Comment (cameel) [inline libsolidity/formal/SymbolicState.cpp:123]: This is a weird use of `move()`. It would make sense if `==` was defined to take arguments by value rather than by reference - but why would one ever do that for `==`?

File: libyul/backends/evm/EVMDialect.cpp [hunk 3]
```diff
@@ -82,7 +82,7 @@ pair<YulString, BuiltinFunctionForEVM> createEVMFunction(
 		_assembly.appendInstruction(_instruction);
 	};
 
-	return {f.name, move(f)};
+	return {f.name, std::move(f)};
```
Comment (cameel) [inline libyul/backends/evm/EVMDialect.cpp]: This whole expression looks like it depends on the order of evaluation of arguments, which is undefined in C++.

File: scripts/ci/build_emscripten.sh [hunk 4]
```diff
@@ -63,6 +63,7 @@ fi
 
 mkdir -p "$BUILD_DIR"
 cd "$BUILD_DIR"
+echo $BUILD_DIR
```
Comment (cameel) [inline scripts/ci/build_emscripten.sh]: Looks unrelated to the PR.

File: libsolidity/formal/SymbolicState.cpp (hunk 2 above)
Comment (Marenz) [reply libsolidity/formal/SymbolicState.cpp:123]: It seems it's because the parameters are moved into the returned object..
```
	friend Expression operator==(Expression _a, Expression _b)
	{
		smtAssert(_a.sort->kind == _b.sort->kind, "Trying to create an 'equal' expression with different sorts");
		return Expression("=", std::move(_a), std::move(_b), Kind::Bool);
	}
```

Comment (cameel) [reply libsolidity/formal/SymbolicState.cpp:123]: Hmm... interesting. I guess it's good then.

Comment (Marenz) [reply libsolidity/formal/SymbolicState.cpp:123]: I agree that it seems an unexpected way to use `operator==`

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 5]
```diff
@@ -236,8 +236,8 @@ string IRGenerator::generate(
 
 	solAssert(_contract.annotation().creationCallGraph->get() != nullptr, "");
 	solAssert(_contract.annotation().deployedCallGraph->get() != nullptr, "");
-	verifyCallGraph(collectReachableCallables(**_contract.annotation().creationCallGraph), move(creationFunctionList));
-	verifyCallGraph(collectReachableCallables(**_contract.annotation().deployedCallGraph), move(deployedFunctionList));
+	verifyCallGraph(collectReachableCallables(**_contract.annotation().creationCallGraph), std::move(creationFunctionList));
+	verifyCallGraph(collectReachableCallables(**_contract.annotation().deployedCallGraph), std::move(deployedFunctionList));
```
Comment (nishant-sachdeva) [inline libsolidity/codegen/ir/IRGenerator.cpp:240]:
```suggestion
	verifyCallGraph(
	     collectReachableCallables(**_contract.annotation().creationCallGraph),
	     std::move(creationFunctionList)
	);
	verifyCallGraph(
	     collectReachableCallables(**_contract.annotation().deployedCallGraph), 
             std::move(deployedFunctionList)
        );
```
Just a coding style change.

File: libsolidity/formal/SMTEncoder.cpp [hunk 6]
```diff
@@ -2991,7 +2991,7 @@ set<FunctionDefinition const*, ASTNode::CompareByID> const& SMTEncoder::contract
 			for (auto const* baseFun: base->definedFunctions())
 				allFunctions.insert(baseFun);
 
-		m_contractFunctionsWithoutVirtual.emplace(&_contract, move(allFunctions));
+		m_contractFunctionsWithoutVirtual.emplace(&_contract, std::move(allFunctions));
```
Comment (nishant-sachdeva) [inline libsolidity/formal/SMTEncoder.cpp:2994]:
```suggestion
		m_contractFunctionsWithoutVirtual.emplace(
		     &_contract, 
		     std::move(allFunctions)
		);
```

File: libsolidity/formal/SymbolicState.cpp [hunk 7]
```diff
@@ -94,12 +94,12 @@ smtutil::Expression SymbolicState::balance() const
...
 smtutil::Expression SymbolicState::balance(smtutil::Expression _address) const
 {
-	return smtutil::Expression::select(balances(), move(_address));
+	return smtutil::Expression::select(balances(), std::move(_address));
 }
 
 smtutil::Expression SymbolicState::blockhash(smtutil::Expression _blockNumber) const
 {
-	return smtutil::Expression::select(m_tx.member("blockhash"), move(_blockNumber));
+	return smtutil::Expression::select(m_tx.member("blockhash"), std::move(_blockNumber));
```
Comment (nishant-sachdeva) [inline libsolidity/formal/SymbolicState.cpp:102]:
```suggestion
	return smtutil::Expression::select(
	     m_tx.member("blockhash"), 
	     std::move(_blockNumber)
	);
```

**Review by nishant-sachdeva — COMMENTED** [review]
Apart from the comments raised by @cameel , and a few minor coding style changes, LGTM

**Comment by leonardoalt** [comment]
The style changes are independent from this PR, I think they should be handled separately.

**Comment by leonardoalt** [comment]
My only question here is: why?

**Comment by Marenz** [comment]
> My only question here is: why?

See PR description...

**Comment by leonardoalt** [comment]
> > My only question here is: why?
> 
> See PR description...

:grimacing: thanks!

File: libyul/backends/evm/EVMDialect.cpp (hunk 3 above)
Comment (Marenz) [reply libyul/backends/evm/EVMDialect.cpp]: https://github.com/ethereum/solidity/pull/13449

**Review by leonardoalt — APPROVED** [review]

File: libsolidity/formal/SymbolicState.cpp (hunk 7 above)
Comment (nikola-matic) [reply libsolidity/formal/SymbolicState.cpp:102]: I'd argue that we shouldn't change the formatting (at least not in this PR), especially if our `check_style` script doesn't complain. I'd much rather prefer to wait until `clang-format` starts supporting everything we want (which should be soon), and then we can just use that to format our code. In addition, it should significantly simplify our `check_style` script, as it should simply do a comparison between current file and `clang-format-X current-file`, and if there's a mismatch, fail the step.

**Review by nikola-matic — APPROVED** [review]
